File: backend/app/schemas/recommendations.py
from typing import Literal

from pydantic import BaseModel


class Recommendation(BaseModel):
    id: str
    title: str
    description: str
    rationale: str
    cost: str
    benefit: str
    risk: str
    costIcon: str
    benefitIcon: str
    riskIcon: str
    priority: Literal["high", "medium", "low"]
    timeframe: Literal["today", "week", "season"]

File: src/api/recommendations.ts
import { apiFetch } from './client';

export interface Recommendation {
  id: string;
  title: string;
  description: string;
  rationale: string;
  cost: string;
  benefit: string;
  risk: string;
  costIcon: string;
  benefitIcon: string;
  riskIcon: string;
  priority: 'high' | 'medium' | 'low';
  timeframe: 'today' | 'week' | 'season';
}

export async function getTodayRecommendation(): Promise<Recommendation> {
  return apiFetch<Recommendation>('/recommendations/today');
}

export async function getWeeklyPlan(): Promise<Recommendation[]> {
  return apiFetch<Recommendation[]>('/recommendations/weekly');
}

export async function getSeasonalPlan(): Promise<Recommendation[]> {
  return apiFetch<Recommendation[]>('/recommendations/seasonal');
}
